frontend/sozisel-app/src/contexts/Quiz/quizReducer.ts

```diff
@@ -0,0 +1,229 @@
+import { Answer, QuizQuestion } from "../../model/Template";
+
+import { v4 as uuidv4 } from "uuid";
+
+export type QuizActions =
+  | { type: "ADD_ANSWER"; answer: Answer; question: QuizQuestion }
+  | { type: "SET_DURATION_TIME"; durationTime: number }
+  | { type: "SET_PERCENTAGE_OF_PARTICIPANTS"; percentageOfParticipants: number }
+  | { type: "SET_TRACKING_MODE"; trackingMode: boolean }
+  | { type: "SET_QUESTIONS"; questions: QuizQuestion[] }
+  | { type: "RESET" }
+  | { type: "DELETE_ANSWER"; answer: Answer; question: QuizQuestion }
+  | { type: "UPDATE_ANSWER"; answer: Answer; question: QuizQuestion }
+  | {
+      type: "TOGGLE_CORRECT_ANSWER";
+      correctAnswer: Answer;
+      question: QuizQuestion;
+    }
+  | {
+      type: "DELETE_CORRECT_ANSWER";
+      correctAnswer: Answer;
+      question: QuizQuestion;
+    }
+  | { type: "ADD_QUESTION"; question: QuizQuestion }
+  | { type: "UPDATE_QUESTION"; question: QuizQuestion }
+  | { type: "DELETE_QUESTION"; question: QuizQuestion };
+
+export interface QuizStoreInterface {
+  questions: QuizQuestion[];
+  durationTime: number;
+  percentageOfParticipants: number;
+  trackingMode: boolean;
+}
+
+export const initialQuestion = {
+  id: uuidv4(),
+  question: "Treść pytania...",
```

Comment:
Add a `// FIXME: get rid of hard coded translation`. It would be a hussle to change it right now but it looks bad.